src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,412 @@
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        const auto& eRequest = std::get<RequestType>(res);
+
+        const auto errorMsg = !eRequest.has_name() ? std::make_optional("Missing /name field") : std::nullopt;
+        if (errorMsg.has_value())
+        {
+            return genericError<ResponseType>(errorMsg.value());
+        }
+
+        auto& sessionManager = SessionManager::getInstance();
+        const auto session = sessionManager.getSession(eRequest.name());
+        if (!session.has_value())
+        {
+            return genericError<ResponseType>(fmt::format("Session '{}' could not be found", eRequest.name()));
+        }
+
+        ResponseType eResponse;
+
+        // TODO: improve creation date representation
+        eResponse.set_creationdate(std::to_string(session->getCreationDate()));
+        eResponse.set_id(session->getSessionID());
+        eResponse.set_lifespan(session->getLifespan());
+        eResponse.set_policyname(session->getPolicyName());
+        eResponse.set_routename(session->getRouteName());
+
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+
+        return toWazuhResponse(eResponse);
+    };
+}
+
+api::Handler sessionsGet(void)
+{
+    return [](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::SessionsGet_Request;
+        using ResponseType = eTest::SessionsGet_Response;
+        auto res = fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        const auto& eRequest = std::get<RequestType>(res);
+
+        auto& sessionManager = SessionManager::getInstance();
+        auto list = sessionManager.getSessionsList();
+
+        ResponseType eResponse;
+        for (const auto& sessionName : list)
+        {
+            eResponse.add_list(sessionName);
+        }
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+
+        return toWazuhResponse(eResponse);
+    };
+}
+
+api::Handler sessionsDelete(shared_ptr<::router::Router> router, shared_ptr<catalog::Catalog> catalog)
+{
+    return [router, catalog](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::SessionsDelete_Request;
+        using ResponseType = eTest::SessionsDelete_Response;
+        auto res = fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        const auto& eRequest = std::get<RequestType>(res);
+
+        const auto errorMsg =
+            !eRequest.has_name() && !eRequest.has_removeall()
```

Comment:
**thought:** Wouldn't it be safer to enclose the logic expression ` !eRequest.has_name() && !eRequest.has_removeall()` in parentheses before applying the conditional operator `?` ?